lua/metals/config.lua

```diff
@@ -37,24 +37,45 @@ local function in_disabled_mode(config)
   return disabled
 end
 
+--- Used to generate Metals installation file name depending on configuration
+--- @return string fileame
+local function metals_filename()
```

Comment:
```suggestion
local function metals_executable_name()
```
If now we'll get warnings about a redefined local down below where we also are defined a `metals_filename`. So we can either change it there or change it here, but let's make them different.

Reply:
This function ends up pretty much returning exactly what `metals_bin()` returns, but I see you use this now in the install instead of `metals_bin()`. Aren't we able to just re-use that?

Reply:
metals_bin returns the actual binary, this function returns the installation name for metals, perhaps name can be improved `metals_install_name` or something, but they are not the same on windows.

Reply:
Ahh cause even on windows you want to install metals, but you end up with `metals.bat`. Got it.